card/src/main/java/com/adyen/checkout/card/ui/AddressFormInput.kt

```diff
@@ -188,20 +200,22 @@ class AddressFormInput @JvmOverloads constructor(
                 @Suppress("UNUSED_VALUE")
                 isErrorFocused = true
                 textInputLayoutProvinceTerritory?.requestFocus()
+                textInputLayoutState?.requestFocus()
             }
             textInputLayoutProvinceTerritory?.showError(localizedContext.getString(provinceTerritoryValidation.reason))
+            textInputLayoutState?.showError(localizedContext.getString(provinceTerritoryValidation.reason))
         }
     }
 
     private fun updateCountries(countryList: List<AddressListItem>) {
         countryAdapter.setItems(countryList)
-        countryList.firstOrNull { it.selected }?.let {
-            val selectedSpecification = AddressSpecification.fromString(it.code)
-            if (formContainer.childCount == 0) {
-                autoCompleteTextViewCountry.setText(it.name)
-                populateFormFields(selectedSpecification)
-            }
+        val selectedCountry = countryList.firstOrNull { it.selected }
+        val selectedSpecification = AddressSpecification.fromString(selectedCountry?.code)
+        if (selectedSpecification != currentSpec) {
+            currentSpec = selectedSpecification
+            autoCompleteTextViewCountry.setText(selectedCountry?.name)
         }
+        populateFormFields(currentSpec)
```

Comment:
We should populate the fields in the if block, we don't have to do it every time.